python/compare_plots.py

```diff
@@ -442,21 +491,30 @@ def draw_plot(sampleDictRef, sampleDictOther, plotname, logY=True, logX=False, p
 
 # Sample
 samplesRef = []
-samplesRef.append("Y3_M200")
+samplesRef.append("B3mL2_M200")
+samplesRef.append("B3mL2_M700")
+samplesRef.append("B3mL2_M1500")
+#samplesRef.append("Y3_M200")
 #samplesRef.append("Y3_M700")
 #samplesRef.append("Y3_M1500")
 #
 samplesOther = samplesRef
 
-# Open files
-sampleDictRef=get_files(samplesRef,year,inDirRef)
-sampleDictOther=get_files(samplesOther,year,inDirOther)
-# List of plots
-listofplots=["mmumu_sel10_mllinclusive_nBTag1p_MuDetAll"]
+listofplots=[]
+listofplots.append("mmumu_sel10_mllinclusive_nBTag1p_MuDetAll")
+listofplots.append("mmumu_sel10_mllinclusive_nBTag1_MuDetAll")
+listofplots.append("mmumu_sel10_mllinclusive_nBTag2p_MuDetAll")
 toexclude = []
 
-for plot in listofplots:
-    if plot in toexclude:
-        continue
-    draw_plot(sampleDictRef, sampleDictOther,  plot, True, False, False, True, lumi, year)
-    draw_plot(sampleDictRef, sampleDictOther,  plot, False, False, False, True, lumi, year)
+# Loop over variations
+for on,o in enumerate(inDirOther):
+    # Open files
+    for sn,s in enumerate(samplesRef):
+        sampleDictRef=get_files([samplesRef[sn]],year,inDirRef)
+        sampleDictOther=get_files([samplesOther[sn]],year,inDirOther[on])
+        # List of plots
+        for plot in listofplots:
+            if plot in toexclude:
+                continue
+            draw_plot(sampleDictRef, sampleDictOther, legRef, legOther[on], outDir[on], plot, True , False, False, True, lumi, year)
+            draw_plot(sampleDictRef, sampleDictOther, legRef, legOther[on], outDir[on], plot, False, False, False, True, lumi, year)
```

Comment:
I am a bit confused about the output directory here. Is it really a list of different directories? Because I think that L468 implies that there is only one directory. In fact, I would think that outDir should be created based on ```inDirOther``` and not on ```inDirRef```. But it is also highly probable that I have missed something, since the changes in this file are quite extensive.

Reply:
This was a left-over from a different test that I was running.
The output dir list is initialized based on `inDirOther`, then the members of the list are passed in a loop to the `draw_plot` function.